superset/migrations/shared/migrate_viz.py

```diff
@@ -0,0 +1,93 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import json
+from typing import Any, Dict, Optional, Set
+
+
+class MigrateViz:
+    remove_keys: Set[str] = set()
+
+    mapping_keys: Dict[str, str]
+
+    viz_type: str
+
+    def __init__(self, v1_data: str) -> None:
+        self.raw_data = v1_data
+        self.data = json.loads(v1_data)
+
+    def post_processing(self, data: Dict[str, Any]) -> Dict[str, Any]:
+        raise NotImplementedError()
+
+    def migrate(self) -> str:
```

Comment:
It would probably be a good idea to support up and down migrations in this class